homeassistant/components/zamg/sensor.py

```diff
@@ -241,36 +261,68 @@ def setup_platform(
     )
 
 
-class ZamgSensor(SensorEntity):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the ZAMG sensor platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.config_entry is None:
+        coordinator.config_entry = entry
+
+    async_add_entities(
+        ZamgSensor(coordinator, entry.title, description)
+        for description in SENSOR_TYPES
+    )
+
+
+class ZamgSensor(CoordinatorEntity, SensorEntity):
     """Implementation of a ZAMG sensor."""
 
     entity_description: ZamgSensorEntityDescription
 
-    def __init__(self, probe, name, description: ZamgSensorEntityDescription):
+    def __init__(self, coordinator, name, description: ZamgSensorEntityDescription):
         """Initialize the sensor."""
+        super().__init__(coordinator)
         self.entity_description = description
-        self.probe = probe
-        self._attr_name = f"{name} {description.key}"
+        # self.coordinator = coordinator
+        self._attr_name = f"{name} {description.name}"
+        self._attr_unique_id = (
+            f"{coordinator.data.get(CONF_STATION_ID)}_{description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, coordinator.data.get(CONF_STATION_ID))},
+            manufacturer=ATTRIBUTION,
+            configuration_url=MANUFACTURER_URL,
+            name=coordinator.name,
+        )
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return self._attr_device_info
```

Comment:
This is already handled in the Entity base class.

Reply:
right, done!